lambda-buffers-compiler/src/LambdaBuffers/Compiler/ProtoCompat/InfoLess.hs

```diff
@@ -0,0 +1,154 @@
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module LambdaBuffers.Compiler.ProtoCompat.InfoLess (
```

Comment:
I'm thinking this might be an overkill.

Let's consider the situations we need to strip the SourceInfo
1. Indexing on names (VarName, TyName, ConstrName, FieldName etc.)
2. Eq on ProtoCompat.Types.CompilerInput

For the first use case, it's sufficient to have a simple `mkInfoLess` function that defaults the SourceInfo. No generic traversals here.

For the second use case, well the entire `ProtoCompat.Types.CompilerInput` with Maps have a desired Eq instance. So we don't need to strip SourceInfo from everything.

I think we should remove most of this generic handling and keep only the InfoLess newtype.

Reply:
I'm happy with InfoLess like this - it really works for my use case atm. 
If you think it works better please make a PR against it - I genuinely don't mind as long as long as it's as easy as now or easier.